v3/src/components/graph/components/axis-attribute-menu.tsx

```diff
@@ -0,0 +1,67 @@
+import { Menu, MenuItem, MenuList, MenuButton, MenuDivider } from "@chakra-ui/react"
+import React, { CSSProperties } from "react"
+import { useDataSetContext } from "../../../hooks/use-data-set-context"
+import { GraphPlace } from "../models/axis-model"
+import { useGraphLayoutContext} from "../models/graph-layout"
+import t from "../../../utilities/translation/translate"
+import { useOverlayBounds } from "../../../hooks/use-overlay-bounds"
+import { useDataConfigurationContext } from "../hooks/use-data-configuration-context"
+
+interface IProps {
+  place: GraphPlace,
+  target: SVGGElement | null
+  portal: HTMLElement | null
+  onChangeAttribute: (place: GraphPlace, attrId: string) => void
+  onTreatAttributeAs: (place: GraphPlace, attrId: string, treatAs: string) => void
+}
+
+export const AxisAttributeMenu = ({ place, target, portal, onChangeAttribute, onTreatAttributeAs }: IProps ) => {
+  const data = useDataSetContext()
+  const dataConfig = useDataConfigurationContext()
+  const { plotWidth, plotHeight, margin } = useGraphLayoutContext()
+  const role = place === "left" ? "y" : "x"
+  const attrId = dataConfig?.attributeID(role)
+  const attribute = attrId ? data?.attrFromID(attrId) : null
+  const treatAs = attribute?.type === "numeric" ? "categorical" : "numeric"
+  const overlayBounds = useOverlayBounds({target, portal})
+  const buttonStyles: CSSProperties = { position: "absolute", color: "transparent" }
+
+  if (attrId === undefined && place === "bottom"){
```

Comment:
```suggestion
  if (!attrId && place === "bottom"){
```
I would use the simpler test here and avoid explicit mention of `undefined`. Any of the other falsy values would likewise be treated as not having an attribute, e.g. empty string, zero, null, etc.